driver_cpl/driver/cesm_comp_mod.F90

```diff
@@ -2050,13 +2056,16 @@ end subroutine cesm_init
  !===============================================================================
 
  subroutine cesm_run()
-   use seq_comm_mct, only : atm_layout, lnd_layout, ice_layout, glc_layout, rof_layout, &
+   use seq_comm_mct,   only: atm_layout, lnd_layout, ice_layout, glc_layout, rof_layout, &
          ocn_layout, wav_layout, esp_layout
+   use shr_string_mod, only: shr_string_listGetIndexF
 
```

Comment:
I don't understand your comment "We've tried to encode this interaction in xml now". My thought was to point anyone looking at user_nl_cpl to the correct interface. Do you have a better suggestion?
An implicit none at module scope applies to "applies to it, its module subprograms, and their internal subprograms" so the other ones are redundant. I removed all of them.